### DISCUSSION TITLE
How to use cordova-admob-pro plugin with Quasar2/Vue3?

### DISCUSSION THREAD

**Discussion by david-katona** [quasarframework/quasar#10534] [CLI - Capacitor / Cordova modes] [answered]
Hi,

I'm trying to make [https://github.com/floatinghotpot/cordova-admob-pro](https://github.com/floatinghotpot/cordova-admob-pro) plugin work in my application but I think there's some missing link so I ask for your help.

I followed the steps from [https://quasar.dev/quasar-cli/developing-cordova-apps/preparation](https://quasar.dev/quasar-cli/developing-cordova-apps/preparation) so I have cordova set up. With `quasar dev -m android` I can run my app both in emulator and on physical device successfully.

Then I installed the admob plugin with `cordova plugin add cordova-plugin-admobpro` following the plugin Quick start guide. Apparently it also succeeded. However my app fails to load because the `AdMob` object is undefined.

How can I access the `AdMob` object from my app? Should it be available in the component `setup()` method? Or from the component template? Or from a boot file? If so how? Or is there any missing steps I should register the plugin?

Thank you,
David

**Comment by ontwikkelfabriek** [comment, marked as answer]
@david-katona 

Have you realized that the admob plugin you are referring to has not been updated in 2 year?

Here's a plugin for Capacitor and Cordova that is maintained:
https://admob-plus.github.io/

Cordova install doc:
https://admob-plus.github.io/docs/cordova/installation

**Reply by david-katona** [reply]
Thank you @ontwikkelfabriek I'll check this admob-plus out.

(Yes I saw it's old but I found it in the comment section of a very recent Quasar mobile app YouTube video so I was hoping it still works.)

**Reply by ontwikkelfabriek** [reply]
@david-katona 

>  so I was hoping it still works

It might still work

**Reply by david-katona** [reply]
Thanks again, the newer one worked OOB, I could display a test interstitial.
The only caveat is that it throws a `TypeError: Cannot read property 'length' of undefined` (coming from node_modules) after the app starts but this might be due to my project configuration. So probably I can suppress it.

I'll check out Qodo, too.

**Reply by capslockoff** [reply]
@david-katona I am running into some issues getting the admob-plus to work with cordova. I am running into some kind of dependency issue.

Do you mind sharing your `android/app/build.gradle` dependencies?

Mine is like this and I have a feeling something is a mismatch

```
    implementation "com.google.android.gms:play-services-auth:15.0.1"
    implementation "com.google.android.gms:play-services-identity:15.0.1"
    implementation "com.google.android.gms:play-services-ads:20.4.0"
```

```
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives:
  Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
  Program type already present: com.google.android.gms.common.api.internal.zzc
```

**Comment by ontwikkelfabriek** [comment]
@david-katona 

Btw here's a github with a very nice project that uses admob:
https://github.com/vycoder/qodo

**Comment by ASQUAREEE** [comment]
I'm continuing to struggle with admob plus cordova; can somebody provide a repository or show me how to call cordova plugins into a quasar vue component?  @david-katona  @ontwikkelfabriek  @capslockoff  @steveclarke
